torch/testing/_internal/jit_utils.py

```diff
@@ -527,6 +528,12 @@ def input_reduce(input, fn, acc):
 
         return ge
 
+    def checkTracerWarning(self, *args, **kwargs):
```

Comment:
Instead of having this function, can we use `unittest.TestCase.assertWarnsRegex`? e.g. all existing callsites can be mechanically replaced like:

```
self.checkTracerWarning(foo, torch.rand(3, 4), _force_outplace=True)
```

to

```
with self.assertWarnsRegex(torch.jit.TracerWarning, "cause the trace to be incorrect""):
  torch.jit.trace(foo, torch.rand(3, 4), _force_outplace=True)
```

This will have the benefit that it's more explicit about what's going on and will force the writer to think about if there is a more specific string they could match on, helping to alleviate the risk of false-positives